examples/solvers/functional_basic_example.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Basic example on how to write a functional.
+
+When defining a new functional, there are a few standard methods and properties
+that can be implemeted. These are:
+- ``__init__``. This method intialize the functional
+- ``_call``. The actual function call ``functional(x)``
+- ``gradient`` (property). This gives the gradient operator of the functional,
+    i.e., the operator that corresponds to the mapping ``x -> grad_f(x)``
+- ``proximal``. This returns the proximal operator. If called only as
+    ``functional.proximal``, it corresponds to a `Proximal factory`.
+- ``conjugate_functional`` (property). This gives the convex conjugate
+    functional
+- ``derivative``. This returns the (directional) derivative operator in a point
+    y, such that when called with a point x it corresponds to the linear
+    operator ``x --> <x, grad_f(y)>``. Note that this has a default
+    implemetation that uses the gradient in order to achieve this.
```

Comment:
Yeah sure, work in progress, but in a different place now. However, the example still is really complex, with an extra property and an inner class. If users see this and read "basic" they probably think "whoa, if this is basic, I don't want to see complex code".
So I suggest that the functional itself (though being mathematically simple), should be really embarrassingly simple, like only `<x, y>` or only `||x||^2`, that should reduce the code to implement the functional and all of its properties.
Then, we don't need test-like code below, just some printouts of the values of the functional and its properties. Shouldn't take significantly more than 10 lines.
Further, instead of the text above, you can now simply make a link to the in-depth doc (hyperlink to the yet non-existing site on ReadTheDocs).
All of that together should get us closer to 100 lines of code.